chain/pool/txpool.go

```diff
@@ -330,14 +345,26 @@ func (tp *TxnPool) CleanSubmittedTransactions(txns []*transaction.Transaction) e
 		if _, ok := tp.TxMap.Load(txn.Hash()); ok {
 			txnsInPool = append(txnsInPool, txn)
 		}
-		tp.TxMap.Delete(txn.Hash())
-		tp.TxShortHashMap.Delete(shortHashToKey(txn.ShortHash(config.ShortHashSalt, config.ShortHashSize)))
+		tp.deleteTransactionFromMap(txn)
 	}
 
-	if err := tp.blockValidationState.CleanSubmittedTransactions(txnsInPool); err != nil {
-		if err := tp.blockValidationState.RefreshBlockValidationState(tp.GetAllTransactions()); err != nil {
-			return err
-		}
+	return tp.CleanBlockValidationState(txnsInPool)
+}
+
+func (tp *TxnPool) addTransactionToMap(txn *transaction.Transaction) {
```

Comment:
Do you think we should put tp.NanoPayTxs.Store here?

Reply:
this is used for all kinds of txs, so I don't think so